common/quotas/limiter_test.go

```diff
@@ -94,6 +94,31 @@ func TestMultiStageRateLimitingMultipleDomains(t *testing.T) {
 	check(" after refill")
 }
 
+func TestMultiStageRateLimitingUserAndWorkerRps(t *testing.T) {
+	t.Parallel()
+	// 2-global allows the 1-domain requests to be consumed while leaving the same remaining limit for global,
+	// so the same checks can be re-performed with the global limit.
+	policy := newFixedRpsMultiStageRateLimiter(2, 1)
+	checkDomain := func(domain, suffix string) {
+		// user bucket allows 1
+		assert.True(t, policy.Allow(Info{Domain: domain, IsUser: true}), "1:1 user should work"+suffix)
+		assert.False(t, policy.Allow(Info{Domain: domain, IsUser: true}), "second 1:1 user should be limited"+suffix)
+		// worker bucket independently allows 1
+		assert.True(t, policy.Allow(Info{Domain: domain, IsUser: false}), "1:1 worker should work despite user being limited"+suffix)
+		assert.False(t, policy.Allow(Info{Domain: domain, IsUser: false}), "second 1:1 worker should be limited"+suffix)
+	}
+	check := func(suffix string) {
+		// per-domain has one set of limits
+		checkDomain("withdomain", " for domains"+suffix)
+		// global has an independent set of limits
+		checkDomain("", " for global"+suffix)
+	}
+
+	check("")
+	time.Sleep(time.Second)
```

Comment:
Can we use mocked clock here? For example this one: https://github.com/benbjohnson/clock

Reply:
we have mock clocks in other packages, but not for this - the `golang.org/x/time.Limiter` uses `time.Now()` internally and we have no way to change that.

so unfortunately no.  but this doesn't slow down this package by 1 second, since they're all run in parallel.

Reply:
and with the rewrite to just use two limiters: this goes away anyway.